From 2bf13a906f43a786de358060f5c764def5211270 Mon Sep 17 00:00:00 2001
From: Carter Chen <carter.chen@infineon.com>
Date: Tue, 28 Mar 2023 22:12:26 -0500
Subject: [PATCH 184/208] brcmfmac: Avoid adding two sets of HE Capab and Oper
 IE, HE 6G Capab to the outgoing mgmt frames

fixing wrong element id
added skip HE 6G capab id

Fixes: SWWLAN-143520
Signed-off-by: Carter Chen <carter.chen@infineon.com>
---
 .../net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c    | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index c46758541343..33f99807a338 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -5238,13 +5238,14 @@ brcmf_parse_extension_ies(const u8 *extension_ie_buf, u32 extension_ie_len,
 			goto next;
 		}
 
-		/* skip parsing the HE capab & oper IE from upper layer
+		/* skip parsing the HE capab, HE_6G_capa & oper IE from upper layer
 		 * to avoid sending it to the FW, as these IEs will be
 		 * added by the FW based on the MAC & PHY capab if HE
 		 * is enabled.
 		 */
-		if (ext_ie->id == WLAN_EID_EXT_HE_CAPABILITY ||
-		    ext_ie->id == WLAN_EID_EXT_HE_OPERATION)
+		if (ext_ie->ext_id == WLAN_EID_EXT_HE_CAPABILITY ||
+		    ext_ie->ext_id == WLAN_EID_EXT_HE_OPERATION ||
+		    ext_ie->ext_id == WLAN_EID_EXT_HE_6GHZ_CAPA)
 			goto next;
 
 		parsed_info = &extension_ies->ie_info[extension_ies->count];
-- 
2.17.1

